tests/Unit/EmogrifierTest.php

```diff
@@ -690,6 +690,118 @@ public function emogrifyNotAppliesCssToNonMatchingElements($css, $expectedHtml)
         static::assertContains(sprintf($expectedHtml, $cssDeclaration1, $cssDeclaration2), $result);
     }
 
+    /**
+     * @return string[][]
+     */
+    public function differentCssSelectorSpecificityDataProvider()
+    {
+        return [
+            'class more specific than types' => ['html body p', '.p-1', '<p class="p-1"'],
+            'ID more specific than class and types' => ['html body .p-4', '#p4', '<p class="p-4" id="p4"'],
+            'ID more specific than 20 classes and types' => [
```

Comment:
Why exactly 20? Could this also be a different number? Maybe a comment would be helpful why this number is here.

Reply:
20 was arbitrary.  Have bumped this to 99 which is the maximum supported (without rewriting the comparator, and I don't think likely to be a real-world limitation).  Also added a comment suggesting a similar test for classes vs 99 types, which would require support for chaining pseudo-classes, e.g. `:not(h1):not(h1)...`